mlir/include/mlir/IR/BuiltinTypeInterfaces.td

```diff
@@ -221,7 +221,17 @@ def ShapedTypeInterface : TypeInterface<"ShapedType"> {
 
     /// Whether the given shape has any size that indicates a dynamic dimension.
     static bool isDynamicShape(ArrayRef<int64_t> dSizes) {
-      return any_of(dSizes, [](int64_t dSize) { return isDynamic(dSize); });
+      return llvm::any_of(dSizes, isDynamic);
+    }
+
+    /// Whether the given dimension size indicates a statically-sized dimension.
+    static constexpr bool isStatic(int64_t dValue) {
+      return !isDynamic(dValue);
```

Comment:
Given that `isDynamic`/`isStatic` are frequently used, wouldn't it be better to avoid this indirection and do `return dValue != kDynamic;` instead?

Reply:
I was actually thinking of potential perf impact. Not asking you to measure, just thinking that inlining that condition would be equally good or possibly better, no?